Program.cs

```diff
@@ -3,46 +3,61 @@
 using System.Linq;
 using System.Diagnostics;
 using Newtonsoft.Json;
+using Microsoft.Extensions.Logging;
 
-namespace dropdownloadcore
+namespace DropDownloadCore
 {
-    class Program
+    sealed class Program
     {
+        private const string RelavePathEnvironmentVariable = "RELATIVEPATH";
+        private const string VSTSPatEnvironmentVariable = "VSTSPAT";
+        private const string DropDestinationEnvironmentVariable = "DROPDESTINATION";
+        private const string DefaultDropDestination = "/drop";
+        private const string DropUrlEnvironmentVariable = "DROPURL";
+
+        private static ILoggerFactory loggerFactory;
+        private static ILogger logger;
+
         static void Main(string[] args)
         {
+            loggerFactory = new LoggerFactory();
+            logger = loggerFactory.CreateLogger<Program>();
+            loggerFactory.AddConsole(LogLevel.Debug); // for now emit all logging info, TODO: make configurable?
```

Comment:
So this will get us times and thread ids attached to output and later we can add other loggers/file/etc. Anything else you had in in mind?

Reply:
Fine to start, I just wanted to try wiring up Microsoft.extensions.logging outside asp.net core to be honest. Can evolve based on need.

Reply:
lets see what it looks like